File: 0977-squares-of-a-sorted-array/0977-squares-of-a-sorted-array.py
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        left = 0
        right = len(nums) -1
        ans = [0] * (right +1)
        ans_pointer = right
        while left <= right:
            if nums[left] * nums[left] >= nums[right] * nums[right]:
                ans[ans_pointer] = nums[left] * nums[left]
                left += 1
            else:
                ans[ans_pointer] = nums[right] * nums[right]
                right -= 1
            ans_pointer -= 1
        return ans

File: 0977-squares-of-a-sorted-array/0977-squares-of-a-sorted-array.java
class Solution {
    public int[] sortedSquares(int[] nums) {
        int[] ans = new int[nums.length];
        int left = 0, right = nums.length -1;
        int ind = nums.length -1;
        while (left <= right) {
            int left_pow = nums[left] * nums[left];
            int right_pow = nums[right] * nums[right];
            if (left_pow > right_pow) {
                ans[ind] = left_pow;
                left ++;
            } else {
                ans[ind] = right_pow;
                right --;
            }
            ind --;
        }
        return ans;
    }
}